xlators/cluster/afr/src/afr-common.c

```diff
@@ -4081,7 +4081,7 @@ _afr_cleanup_fd_ctx(xlator_t *this, afr_fd_ctx_t *fd_ctx)
     }
     GF_FREE(fd_ctx->opened_on);
     GF_FREE(fd_ctx);
-    return;
+    fd_ctx = NULL;
```

Comment:
`fd_ctx` is a local variable. Assigning it to NULL here doesn't prevent the use-after-free detected by coverity (`fd_ctx` will still have the previous value in line 4157).

Reply:
Ugh, you're right.

Reply:
You're right. My bad.
Initially I fixed the issue like this https://github.com/gluster/glusterfs/commit/d437e8ef69d22537fd42525515af929d09bc8cdd, but then I though it's better to move the cleaning into the method itself, as it's more appropriate on an engineering level that the full clean will be done inside.
I'll post a fix soon.

Reply:
I don't fully agree... `_afr_cleanup_fd_ctx()` is like a destructor similar to other memory management functions. In all other places it's the caller who needs to be sure that the pointer is not used any more after calling the destructor.

In this particular case, following the same idea, `GF_FREE()` should be the responsible of clearing the pointer, but we don't use this approach.

For consistency with other memory management functions, I think we should use the same approach here.

@pranithk @amarts what do you think ?